server/server/api/views/views.py

```diff
@@ -1010,3 +1012,24 @@ def load_json(path):
         data_dir = current_app.config.get('DATA_REP_DIR') / 'sc_bilara_data'
         with (data_dir / path).open() as f:
             return json.load(f)
+
+
+class SuttaFullPath(Resource):
+    @cache.cached(key_prefix=make_cache_key, timeout=default_cache_timeout)
+    def get(self, uid):
+        db = get_db()
+
+        full_path = []
+
+        for level in range(1, 10):
+            try:
+                results = db.aql.execute(SUTTA_PARENT, bind_vars={'uid': uid, 'level': level}).next()
+                if results:
+                    full_path.append(results)
+            except StopIteration:
+                break
+
+        full_path.append('pitaka')
+        return '/' + '/'.join(list(reversed(full_path)))
```

Comment:
I think the logic of going up through the parents and getting the path we want can be implemented completely in AQL.
I think it would be more correct to try to implement it